pkg/cluster/fixetcd.go

```diff
@@ -0,0 +1,517 @@
+	if err != nil {
+		return nil, nil, nil, nil, err
+	}
+	scc.ObjectMeta = metav1.ObjectMeta{
+		Name: name,
+	}
+	scc.Groups = []string{}
+	scc.Users = []string{usersAccount}
+	scc.Namespace = nameSpace
+
+	scc, err = m.securitycli.SecurityV1().SecurityContextConstraints().Create(ctx, scc, metav1.CreateOptions{})
+	if err != nil {
+		return nil, nil, nil, nil, err
+	}
+
+	return scc, cr, rb, serviceAcc, nil
+}
+
+func (m *manager) backupEtcdData(ctx context.Context, node string) error {
+	b, err := m.kubernetescli.BatchV1().Jobs(nameSpace).Create(ctx, &batchv1.Job{
+		TypeMeta: metav1.TypeMeta{},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      jobName,
+			Namespace: nameSpace,
+			Labels:    map[string]string{"app": jobName},
+		},
+		Spec: batchv1.JobSpec{
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      jobName,
+					Namespace: nameSpace,
+					Labels:    map[string]string{"app": jobName},
+				},
+				Spec: corev1.PodSpec{
+					RestartPolicy: corev1.RestartPolicyOnFailure,
+					NodeName:      node,
+					Containers: []corev1.Container{
+						{
+							Name:  "backup",
+							Image: image,
+							Command: []string{
+								"chroot",
+								"/host",
+								"/bin/bash",
+								"-c",
+								shellScriptFixEtcd,
+							},
+							VolumeMounts: []corev1.VolumeMount{
+								{
+									Name:      "host",
+									MountPath: "/host",
+									ReadOnly:  false,
+								},
+							},
+							SecurityContext: &corev1.SecurityContext{
+								Capabilities: &corev1.Capabilities{
+									Add: []corev1.Capability{"SYS_CHROOT"},
+								},
+								Privileged: to.BoolPtr(true),
+							},
+							Env: []corev1.EnvVar{
+								{
+									Name:  "BACKUP",
+									Value: "true",
+								},
+							},
+						},
+					},
+					Volumes: []corev1.Volume{
+						{
+							Name: "host",
+							VolumeSource: corev1.VolumeSource{
+								HostPath: &corev1.HostPathVolumeSource{
+									Path: "/",
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}, metav1.CreateOptions{})
+	if err != nil {
+		return err
+	}
+
+	w, err := m.kubernetescli.BatchV1().Jobs(b.Namespace).Watch(ctx, metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+	time.Sleep(30 * time.Second)
+
+	wc := w.ResultChan()
+	<-wc
+
+	m.kubernetescli.BatchV1().Jobs(b.Namespace).Delete(ctx, jobName, metav1.DeleteOptions{})
+	w.Stop()
+
+	return nil
+}
+
+func (m *manager) checkEtcds(ctx context.Context, e *operatorv1.Etcd) bool {
+	for _, c := range e.Status.Conditions {
+		if c.Type == etcdEndpointsDegraded && c.Status == operatorv1.ConditionTrue {
+			return true
+		}
+	}
+	return false
+}
+
+func (m *manager) comparePodEnvToIP(pods *corev1.PodList) *degradedEtcd {
+	de := &degradedEtcd{}
+	for _, p := range pods.Items {
+		etcdIP := m.ipFromEnv(p.Spec.Containers, p.Name)
+		for _, ip := range p.Status.PodIPs {
+			if ip.IP != etcdIP && etcdIP != "" {
```

Comment:
This is where where the decision to move forward or not is made. On line 66 if the struct is empty we exit.
I'll modify this to return an error if the struct is empty.

Reply:
This topic is still open, but I think we can put a pin in it for now since we haven't been able to find a reproducer, and your code change bails out of the remediation in this scenario (for now).